src/backend/distributed/README.md

```diff
@@ -1907,6 +1908,29 @@ Citus relies fully on Postgres to enforce foreign keys. To provide that, Citus r
 
+
+## DROP TABLE
+
+Citus' handling of `DROP TABLE` is slightly different than other DDL operations. In this section, we aim to highlight the key differences and their reasoning.
+
+Citus implements an event trigger, [`citus_drop_trigger()`](https://github.com/citusdata/citus/blob/main/src/backend/distributed/sql/udfs/citus_drop_trigger/latest.sql). The trigger is defined as:
+```sql
+ select * from pg_event_trigger ;
+┌───────┬────────────────────────────┬──────────┬──────────┬─────────┬────────────┬─────────┐
+│  oid  │          evtname           │ evtevent │ evtowner │ evtfoid │ evtenabled │ evttags │
+├───────┼────────────────────────────┼──────────┼──────────┼─────────┼────────────┼─────────┤
+│ 16676 │ citus_cascade_to_partition │ sql_drop │       10 │   16675 │ O          │         │
+└───────┴────────────────────────────┴──────────┴──────────┴─────────┴────────────┴─────────┘
+(1 row)
+```
+One of the primary reasons for utilizing a trigger for `DROP` processing is that after executing `standardProcess_Utility`, the `oid` of the table being dropped is eliminated from Postgres' catalog tables. This makes it more challenging to manage a dropped table in `PostProcessUtility`, as is customary for many `DDL` commands. Instead, we depend on the event trigger to supply the `oid` of the table that has been dropped. This allows us to delete all related metadata, such as entries in `pg_dist_partition` or `pg_dist_shard` from Citus' catalog tables. Additionally, we eliminate all relevant metadata from every node in the cluster. Ultimately, this enables us to remove the shard placements linked to the dropped Citus table.
+
+The drop trigger proves useful for capturing all tables affected by the `CASCADE` operation. For instance, if you delete a parent table, Postgres will automatically execute a `DROP TABLE` command for its partitions. Citus can then seamlessly apply the same operation to all cascaded tables, eliminating the need for manual identification of tables that would be affected by the cascade.
+
+Generally speaking, there isn't a compelling reason to avoid using `PostProcessUtility` for managing `DROP TABLE` commands. Theoretically, one could implement all the same logic within `PostProcessUtility`. However, the drop trigger offers a convenient approach.
```

Comment:
@marcocitus do you agree with that or am I missing something? I couldn't see anything that'd prevent us doing the same in postprocess.

It seems when this logic was added (enterprise commit: [d7a51239f6ffe6e761e6039c456f07363e7a589f](https://github.com/citusdata/citus-enterprise/commit/d7a51239f6ffe6e761e6039c456f07363e7a589f)) it was useful due to some not-applicable reasons for today's Citus.

Reply:
You'd also need to be careful to handle all possible DROP commands, but in theory it's possible.